### PR TITLE
Admin settings dashboard and modal addition for user components

### LINKED ISSUE #890: modal for admin user pages
### Is your feature request related to a problem? Please describe.

The user page is linked from the admin panel and does not have the look of the newer pages with modals.

### Describe the solution you'd like

The look of these pages will change substantially to align with the new modal card look in OED. The top of the page will have a create button to create a user that is similar to the current create page but as a modal popup. A card for each user will have the email & role listed along with a button to edit the user. The modal that pops up will allow changes to the role and, unlike the current page, should allow for password change. There should also be a button to delete the user similar to the conversion page.

Another change is the "Users" button under "Manage:" on the admin page will be removed. In the dropdown menu in the top, right of the OED page, an additional option will be added that is labeled "Users" (only if an admin). This will take the admin to the user card page. The user card page should have an edit/create button with a cancel/discard as on other admin modal card pages along with similar logic/layout. The delete choice should confirm as on other pages.

A couple of other changes:

- The translation in src/client/app/translations/data.ts for "Manage" is likely to be unused after these changes and can be removed in all languages.
- There should be no default user role and the admin should be required to choose one.
- The current page does not shift to the top as other pages. It is assumed this will be fix automatically when made similar to other pages.

### Additional context

None

### PR DESCRIPTION
**Pull request by hazeltonbw**
# Description

In this PR the main focus was to streamline the admin panel to make it more user friendly and future proof for additional features. 
There were also changes to the components that allow admins to manage users. These changes were made to stay consistent with the design choices from other pages which utilizes modals to create, edit, and delete users. 

Fixes: #890 

## Type of change

- [ ] Note merging this changes the database configuration.
- [x] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request
- [x] You acknowledge that every person contributing to this work has signed the [OED Contributing License Agreement](https://openenergydashboard.github.io/developer/cla.html) and each author is listed in the Description section.

### CODE REVIEW

File: src/client/app/components/admin/AdminSideBar.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (huss) [inline src/client/app/components/admin/AdminSideBar.tsx:1]: While I think this change to a sidebar is interesting, I have a couple of thoughts:

1. This is only done on this one page and I wonder if OED is going to do this should it be part of all the pages that are appropriate such as meters. However, on meters it is a modal so formatting and layout would need to be considered. Given this, I think the project should discuss whether or not to do this.
2. If OED is going to do this generally then we should consider how. For example, react-pro-sidebar is a very fancy one (maybe too much) but packages should be analyzed before we do this via our own styling.

I'm open to ideas but right now I'm inclined to remove this and create something new that addresses this one idea.

Just a few overview comments for the record:

- The items are not alphabetical as are menus in OED.
- Users are being removed from this page.
- Might be nice to have an all option to see everything.

**Comment by juanjoseguva** [comment]
Please add me as a participant to this pull request.

**Comment by juanjoseguva** [comment]
Looking forward to polishing and finalizing this!

File: src/client/app/components/admin/CreateUserModalComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react';
```
Comment (huss) [inline src/client/app/components/admin/CreateUserModalComponent.tsx:5]: I want to understand how this relates to PR #1306 that created cards/modal for users. First, if it is the same basic work then this PR must be held until that work is review/approved in the separate PR. Second, the plan was to remove user from the admin panel (now site settings) so not sure if/where this belongs.

File: src/client/app/components/admin/PreferencesComponent.tsx [hunk 3]
```diff
@@ -16,13 +16,18 @@ import { showErrorNotification, showSuccessNotification } from '../../utils/noti
 import translate from '../../utils/translate';
 import TimeZoneSelect from '../TimeZoneSelect';
 import { defaultAdminState } from '../../redux/slices/adminSlice';
+import UserDetailComponent from './UsersDetailComponent';
 
+interface PreferencesProps {
+	selectedPreference: string;
+}
 
 // TODO: Add warning for invalid data
```
Comment (huss) [inline src/client/app/components/admin/PreferencesComponent.tsx:25]: Status of TODO?

File: src/client/app/components/admin/PreferencesComponent.tsx [hunk 4]
```diff
@@ -16,13 +16,18 @@ import { showErrorNotification, showSuccessNotification } from '../../utils/noti
...
 /**
```
Comment (huss) [inline src/client/app/components/admin/PreferencesComponent.tsx:26]: ESLint warning on missing param in JSDoc.

File: src/client/app/components/admin/UserViewComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-mixed-spaces-and-tabs */
```
Comment (huss) [inline src/client/app/components/admin/UserViewComponent.tsx:1]: Same question from other file on how relates to other PR. Also applies to UsersDetailsComponent.

File: src/client/app/translations/data.ts [hunk 6]
```diff
@@ -503,34 +504,34 @@ const LocaleTranslationData = {
 	},
 	"fr": {
 		"3D": "3D",
-		"400": "400 Bad Request\u{26A1}",
+		"400": "400 Bad Request",
```
Comment (huss) [inline src/client/app/translations/data.ts]: I am unclear on why you removed the special symbol at the end of untranslated strings and seem not to have done for all. Please see the PR that created them and then let me know if you think they should still be removed.

File: src/client/app/translations/data.ts [hunk 7]
```diff
@@ -846,84 +846,86 @@ const LocaleTranslationData = {
...
+		"remove": "Remove",
 		"restore": "Restaurer",
-		"return.dashboard": "Return To Dashboard\u{26A1}",
-		"role": "Role\u{26A1}",
-		"save.all": "Save all\u{26A1}",
-		"save.map.edits": "Save map edits\u{26A1}",
+		"return.dashboard": "Return To Dashboard",
+		"role": "Rôle",
+		"save.all": "Sauver tous",
+		"save.map.edits": "Enregistrer les modifications de la carte",
```
Comment (huss) [inline src/client/app/translations/data.ts]: You seem to have translated strings that are not directly involved in this PR. Is that correct or am I missing something? If so then: First, should this be a separate PR? Second, since you only did a few, is this a language you are very good at so the translations will be of high quality?

**Review by huss — CHANGES_REQUESTED** [review]
Thanks to @hazeltonbw for doing this work.

I have done a high level review of this PR. Given my comments I want to see how they are resolved before doing a complete review as it may lead to nontrivial change. I'm happy to discuss any comments and hear your thoughts.

Here are a couple of high level comments that did not fit anywhere or I could not put in comments in the file:
1. The test build on GitHub is failing because the new translation files do no conform to what OED needs. I suspect they may go away but if not then this must be addressed.
2. I'm unclear on the new en.ts, fr.ts and es.ts translation files. They overlap the data.ts ones. I recall a team meeting where they were discussed and I thought we resolved not to separate out the translations (at least not now). Could you help me understand and let me know if my recollection is off?

**Comment by hazeltonbw** [comment]
This PR has been superseded by PR #1316 and will now be closed.
